docs/concepts/gardenlet.md

```diff
@@ -363,6 +363,34 @@ This internal health status is available via the `gardenlet`'s `/healthz` endpoi
+- `delete`: this flow is triggered when the shoot's `deletionTimestamp` is set, i.e., when it is deleted.
+
+Gardenlet takes special care to prevent unnecessary shoot reconciliations.
+This is important for several reasons, e.g., to not overload the seed API servers and to not exhaust infrastructure rate limits too fast.
+Gardenlet performs shoot reconciliations according to the following rules:
+
+- If `status.observedGeneration` is less than `metadata.generation`: this is the case, e.g., when the spec was changed, a [manual reconciliation operation](../usage/shoot_operations.md) was triggered, or the shoot was deleted.
+- If the [last operation](../usage/shoot_status.md) was not successful.
+- If the shoot is in a [failed state](../usage/shoot_status.md), gardenlet does not perform any reconciliation on the shoot (unless the retry operation was triggered). However, it syncs the `Cluster` resource to the seed in order to inform extension controllers about the failed state.
+- Regular reconciliations are performed with every `GardenletConfiguration.controllers.shoot.syncPeriod` (defaults to `1h`).
+- Shoot reconciliations are not performed if the assigned seed cluster is not healthy or has not been reconciled by the current gardenlet version yet (determined by the `Seed.status.gardener` section). This is done to make sure that shoots are reconciled with fully rolled out seed system components after a gardener upgrade. Otherwise, gardenlet might perform operations of the new version that doesn't match the old version of the deployed seed system components, which might lead to unspecified behavior.
+
+There are a few special cases that overwrite or confine how often and under which circumstances periodic shoot reconciliations are performed:
+
+- In case, the gardenlet config allows it (`controllers.shoot.respectSyncPeriodOverwrite`, disabled by default), the sync period for a shoot can be increased individually by setting the `shoot.gardener.cloud/sync-period` annotation. This is always allowed for shoots in the `garden` namespace. Shoots are not reconciled with a higher frequency than specified in `GardenletConfiguration.controllers.shoot.syncPeriod`.
+- In case, the gardenlet config allows it (`controllers.shoot.respectSyncPeriodOverwrite`, disabled by default), shoots can be marked as "ignored" by setting the `shoot.gardener.cloud/ignore` annotation. In this case, gardenlet does not perform any reconciliation for the shoot.
+- In case `GardenletConfiguration.controllers.shoot.reconcileInMaintenance` is enabled (disabled by default), gardenlet performs regular shoot reconciliations only once in the respective maintenance time window (`GardenletConfiguration.controllers.shoot.syncPeriod` is ignored). Gardenlet randomly distributes shoot reconciliations over the maintenance time window to avoid high bursts of reconciliations (see [this doc](../usage/shoot_maintenance.md#cluster-reconciliation)).
```

Comment:
```suggestion
- In case `GardenletConfiguration.controllers.shoot.reconcileInMaintenanceOnly` is enabled (disabled by default), gardenlet performs regular shoot reconciliations only once in the respective maintenance time window (`GardenletConfiguration.controllers.shoot.syncPeriod` is ignored). Gardenlet randomly distributes shoot reconciliations over the maintenance time window to avoid high bursts of reconciliations (see [this doc](../usage/shoot_maintenance.md#cluster-reconciliation)).
```